src/wazuh_modules/agent_info/agent_info_impl/src/agent_info_impl.cpp

```diff
@@ -706,3 +770,589 @@ nlohmann::json AgentInfoImpl::ecsData(const nlohmann::json& data, const std::str
+    {
+        bool success;              ///< True if operation succeeded (error code 0)
+        std::string response;      ///< Raw response string
+        int errorCode;             ///< Parsed error code (0 if success)
+        bool isModuleUnavailable;  ///< True if error indicates module is unavailable (50-53)
+    };
+
+    /// @brief Parse JSON response and extract error information
+    /// @param responseStr Raw JSON response string
+    /// @return ModuleResponse with parsed information
+    ModuleResponse parseModuleResponse(const std::string& responseStr)
+    {
+        ModuleResponse result;
+        result.response = responseStr;
+        result.success = false;
+        result.errorCode = -1;
+        result.isModuleUnavailable = false;
+
+        try
+        {
+            nlohmann::json responseJson = nlohmann::json::parse(responseStr);
+
+            if (responseJson.contains("error") && responseJson["error"].is_number())
+            {
+                result.errorCode = responseJson["error"].get<int>();
+                result.success = (result.errorCode == 0);
+                result.isModuleUnavailable = MQ_IS_MODULE_UNAVAILABLE(result.errorCode);
+            }
+        }
+        catch (const std::exception&)
+        {
+            // If parsing fails, keep default values
+        }
+
+        return result;
+    }
+
+    /// @brief RAII wrapper for C-style strings allocated with malloc/strdup
+    struct CStringDeleter
+    {
+        void operator()(char* ptr) const
+        {
+            if (ptr)
+            {
+                free(ptr);
+            }
+        }
+    };
+    using UniqueCString = std::unique_ptr<char, CStringDeleter>;
+}
+
+bool AgentInfoImpl::coordinateModules(const std::string& table)
+{
+    // Check if query function is available
+    if (!m_queryModuleFunction)
+    {
+        m_logFunction(LOG_WARNING, "Module query function not available, skipping coordination");
+        return false;
+    }
+
+    // Determine if we should increment version based on table
+    // AGENT_METADATA_TABLE -> increment (max + 1)
+    // AGENT_GROUPS_TABLE -> keep max (no increment)
+    bool incrementVersion = (table == AGENT_METADATA_TABLE);
+
+    // Configuration
+    const std::vector<std::string> modules = {"sca", "syscollector", "fim"};
+    const int MAX_RETRIES = 3;
+    const int RETRY_DELAY_MS = 1000;
+
+    // State tracking
+    std::set<std::string> availableModules;
+    std::set<std::string> pausedModules;
+    std::map<std::string, int> moduleVersions;
+
+    m_logFunction(LOG_DEBUG, "Starting module coordination process");
+
+    std::string agentId, agentName;
+
+    if (m_isAgent)
+    {
+        if (!readClientKeys(agentId, agentName))
+        {
+            m_logFunction(LOG_WARNING, "Failed to read agent name from client.keys, using default");
+            agentName = "unknown";
+        }
+    }
+    else
+    {
+        // For manager, use hostname
+        agentName = "manager";
```

Comment:
What about the hostname?

Reply:
Removed, they were left over from an earlier version of the JSON where the origin field was included.